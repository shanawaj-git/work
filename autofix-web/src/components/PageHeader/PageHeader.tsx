import React, { FC } from "react";
import Section from "@/components/Section";
import { utils } from "@albathanext/design-system";

interface PageHeaderProps {
  children?: JSX.Element | JSX.Element[];
  testId?: string;
  className?: string;
  style?: React.CSSProperties;
}

const PageHeader: FC<PageHeaderProps> = ({
  children,
  testId,
  className,
  style,
  ...props
}: PageHeaderProps) => (
  <Section
    {...props}
    className={"p-6".concat(" ", className || "")}
    data-testid={testId || "PageHeader"}
    style={{
      ...style,
      backgroundColor: utils.colorSelector("P600"),
      minHeight: "fit-content",
      ...style,
    }}
  >
    {children}
  </Section>
);

export default PageHeader;
