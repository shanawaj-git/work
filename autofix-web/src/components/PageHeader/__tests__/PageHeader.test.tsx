import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import PageHeader from "../PageHeader";

describe("<PageHeader />", () => {
  test("it should mount", () => {
    render(<PageHeader />);

    const pageHeader = screen.getByTestId("PageHeader");

    expect(pageHeader).toBeInTheDocument();
  });

  test("it should render child component", () => {
    const text = "I'm a child within header component";
    const pageHeader = render(
      <PageHeader>
        <p data-testid="child">{text}</p>
      </PageHeader>
    );
    const child = pageHeader.getByTestId("child");

    expect(child.innerHTML).toContain(text);
  });

  test("it should match snapshot", () => {
    const { container } = render(<PageHeader />);

    expect(container).toMatchSnapshot();
  });
});
