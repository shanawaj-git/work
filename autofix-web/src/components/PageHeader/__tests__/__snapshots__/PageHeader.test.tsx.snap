// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageHeader /> it should match snapshot 1`] = `
<div>
  <div
    class="min-h-screen p-6 "
    data-testid="PageHeader"
    style="min-height: fit-content;"
  />
</div>
`;
